### ISSUE TITLE
Windows 11 Mac M1 Pro - Pause Error

### ISSUE THREAD

**Issue by stradivarius2a** [utmapp/UTM#3918]
Just installed Windows 11 on my Mac M1 Pro with UTM using the Windows ARM Image from Microsoft.

I have installed Spice Tools, everything is ok except I cannot not "pause" the vm, I have the error message : 
"Failed to save VM snapshot, Usually this means at least one device does not support snapshots. Error : State blocked by non-migratable device ''0000:00:09.0/nvme'.

Can you help me? I didn't change anything on the NVME disk. It was created automatically by the microsoft vm image.

Thanks a lot.

**Comment by conath** [comment]
UTM does not support pausing VMs that use the NVMe interface for one or more drives. Since you have installed the SPICE tools, you might be able to switch the VM’s drive to VirtIO interface and get pause working that way. However Windows might refuse to boot from the VirtIO drive without some [massaging](https://www.diskpart.com/windows-11/repair-windows-11-uefi-bootloader-0725.html).

**Comment by stradivarius2a** [comment]
> UTM does not support pausing VMs that use the NVMe interface for one or more drives. Since you have installed the SPICE tools, you might be able to switch the VM’s drive to VirtIO interface and get pause working that way. However Windows might refuse to boot from the VirtIO drive without some [massaging](https://www.diskpart.com/windows-11/repair-windows-11-uefi-bootloader-0725.html).

Thanks a lot!

**Comment by manishie** [comment]
@stradivarius2a did you have any luck? The "massaging" suggestions didn't work for me with the VirtIO drive...

**Comment by stradivarius2a** [comment]
> @stradivarius2a did you have any luck? The "massaging" suggestions didn't work for me with the VirtIO drive...

I'm still using NVME drive without massaging... sorry....

**Comment by kenny1983** [comment]
While I appreciate you taking the time to reply @conath, your suggestion isn't very helpful, since that link refers to the process for repairing the bootloader on a bare metal Windows install. Could you (or anyone else reading this) please explain in detail how to accomplish this "massaging" technique as you call it?

I have a very similar setup to the OP (MacBook Air M1 instead of Pro) and starting up the VM every morning takes about 10 minutes since 90% of my web developer toolchain has to be x86-emulated. So I'd *really* love to be able to suspend the VM at the end of each work day and start it up within seconds on the next.

MTIA +1

**Comment by conath** [comment]
@kenny1983 The process is the same as on "bare metal". Follow the [massaging](https://www.diskpart.com/windows-11/repair-windows-11-uefi-bootloader-0725.html) tutorial. When it tells you to "Connect your Windows 11 bootable media", insert the install ISO into the VM and boot off it, then use the command line (shift-F10 at the language selection) to re-configure the UEFI bootloader ([massaging](https://www.diskpart.com/windows-11/repair-windows-11-uefi-bootloader-0725.html)).

If you are unhappy with UTM emulation performance, you could [install Windows 11 for ARM in *virtualization* mode](https://mac.getutm.app/gallery/windows-10) instead and use its built-in x86/X64 emulation instead. From what I've heard it's a lot faster this way, though I have not tested this myself. Note Windows 11 ARM doesn't come with VirtIO storage drivers, either.

**Comment by kenny1983** [comment]
Wow thanks for the super fast reply! I apologise for my n00bishness; I tried simply booting off the VM image after changing the storage type which obviously didn't work, and left me on a screen of the Windows recovery environment that I didn't recognise, so I panicked lol.

I might try your suggestion to use virtualisation mode instead over the weekend, since it was a pain to setup my regular toolchain in the first place anyway - only 1 app out of the 5 that I regularly use have an ARM build, and don't get me started on getting IIS working! 😛. It seems as though I should have done more research on this whole "Windows on an M1 Mac" scenario before I even started!

Anyway thanks again for the assistance 😄

**Comment by gfourdx** [comment]
@kenny1983
I solved the problem.

UTM Version 3.24 + Windows 11 Version 22000.586

PS: I'm Chinese and I'm not good at english. please try to understand

solution:
1、download [virtio-win-0.1.221.iso](https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/archive-virtio/virtio-win-0.1.221-1/virtio-win-0.1.221.iso)
2、before installation, right click, click the drive, change NVMe to VirtIO, then save and double click to install windows
3、step by step, you'll see, Windows says there is no disk drive, then mount virtio-win-0.1.221.iso and click load drivers, click D:\viostor\win11\ARM64, enter, next .... , finally, you can see the drive
4、umount virtio-win-0.1.221.iso and mount windows.iso , continue installation
5、the end, mount spice-guest-tools-0.164.3.iso and install, so you can pause

**Comment by kenny1983** [comment]
Wow thank you so much! I've actually switched to using Parallels now so
this has kinda become irrelevant for me now, but I still appreciate you
taking the time to figure out that solution and post it here 😁. Hopefully
it helps someone else who comes across this thread in the future!

**Comment by maadilpm** [comment]
@gfourdx Can you kindly explain the steps in detail? I have the same issue @kenny1983 had. I am not a very technical person.
Before installation meaning just before starting the VM for the first time?
I have tried everything doesn't work. Definitely missing something

**Comment by conath** [comment]
@maadilpm If you want to install from ISO, I can confirm the steps written by @gfourdx. It's similar to my [video tutorial](https://youtu.be/KsNoGPczavw?t=292). You need to create a Windows 11 install ISO, then create a VM and follow the steps. When you get to the installer, follow my video instructions to load the VirtIO storage driver.

If you are using the VMDK method of installing Windows VM, I recommend you leave it at NVMe and don't use the pause feature, sorry. It's very complicated to change to VirtIO after the install.

**Comment by maadilpm** [comment]
@conath I've seen your video, and you are using an emulator for x86 windows. Now obviously emulated windows will be extremely slow when compared to using the native ARM version.

So my question: Can I get an iso file for the ARM version, then this would solve my issues, given i follow your tutorial.

And your 2nd point, do you mean VHDX? If so I don't mind reinstalling it again if it will solve the issue even if it's complicated, worth giving it a try.

**Comment by conath** [comment]
@maadilpm Looks like there are [instructions on the UTM website](https://mac.getutm.app/gallery/windows-10), including how to get the Windows 11 ISO! 
What the website instructions are missing is the whole VirtIO topic. You will want to change the VM config to VirtIO drive interface before you boot the installer, and load the driver during setup like I linked to above.

Regarding 2nd point, with VHDX method. Here are sone vague instructions (sorry don’t have much time): you have to get an ISO anyway to fix the windows bootloader config with `bcdboot` after changing the drive interface. But it would not require reinstall. You would change the drive interface, start the VM and get a BSOD. Shut it down, then insert the Windows ARM ISO, start from it. Go through the setup until the load driver screen and insert the virtio ISO (not guest tools) to load the drivers. Then open CMD and use `bcdboot` to fix the bootloader. Good luck!

**Comment by gfourdx** [comment]
@maadilpm so, What's the problem? Can you describe the problem in detail?

**Comment by bgmarquis** [comment]
I recently moved from UTM w/ NVMe drive to VirtIO...  steps were straightforward.

1) download [virtio-win-0.1.221.iso](https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/archive-virtio/virtio-win-0.1.221-1/virtio-win-0.1.221.iso)
2) install VirtIO-guest-tools-x64
3) shutdown the guest, add a virt-IO disk
4) boot the guest - confirm the guest sees the new virt-IO disk -- if successful, shutdown the guest
5) change the guest boot drive from NVMe to virt-io
6) boot the guest and confirm the guest will suspend correctly.

Cheers!

**Comment by richardschoen** [comment]
Hey all. I just ran into the same situation and turned on Hibernate Mode and now I can hibernate and un-hibernate my Windows 11 instance.  Hope it works for you.

**Comment by fabiradi** [comment]
> I recently moved from UTM w/ NVMe drive to VirtIO... steps were straightforward.
> 
> 1. download [virtio-win-0.1.221.iso](https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/archive-virtio/virtio-win-0.1.221-1/virtio-win-0.1.221.iso)
> 2. install VirtIO-guest-tools-x64
> 3. shutdown the guest, add a virt-IO disk
> 4. boot the guest - confirm the guest sees the new virt-IO disk -- if successful, shutdown the guest
> 5. change the guest boot drive from NVMe to virt-io
> 6. boot the guest and confirm the guest will suspend correctly.
> 
> Cheers!

### 👍🏻 This works great!

Be sure to _not_ make the same mistake that I did:

- **Do not install** ~~`virtio-win-guest-tools`~~: The installation fails.
- **Install instead** `virtIO-guest-tools-x64`. This will install just fine. I even installed _over_ my SPICE guest tools.
  - 🤦🏻 It is clearly stated in the list, but you may also tend to click the wrong file...

**Comment by delphinus81** [comment]
> Hey all. I just ran into the same situation and turned on Hibernate Mode and now I can hibernate and un-hibernate my Windows 11 instance. Hibernate pretty much acts like a Windows shutdown so it hibernates and stops the VM. When you re-start the VM it will un-hibernate and just log back in and your apps are running where they left off. Hope it works for you.

----

Can you share how you make hibernate works? Or your VM setting?

I have iOS installation v4.1.5, arm windows 11.
Disk type Virtio so I can use suspend or pause
(I tried making hibernate works in Nvme, but never able to)

In the beginning after windows 11 installation, hibernate seemed to work. But after windows update, it no longer working

**Comment by fabiradi** [comment]
> Can you share how you make hibernate works? Or your VM setting?

You have to enable it in the control panel. After that, there is a new option when shutting down. See this guide: https://support.microsoft.com/en-us/windows/shut-down-sleep-or-hibernate-your-pc-2941d165-7d0a-a5e8-c5ad-8c972e8e6eff

**Comment by delphinus81** [comment]
> Also: When staring the hibernate mode, be sure to wait for it to actually finish. The screen will stay black for quite some time. Wait for the windows to "blur" and show the "play" button, again.
-----
Thanks, I have enabled the hibernate from Windows, the option is now appearing.

But everytime I hibernate the computer (confirmed until Windows finished hibernating) and switch it back on.
I lost all the open apps and browser that I left... It is as if starting from shutdown condition.

Anyone experiencing same issue?

**Comment by delphinus81** [comment]
> 
@fabiradi , what is your windows arm 11 build installed? I started to feel it is windows and not UTM issue.

**Comment by fabiradi** [comment]
> @fabiradi , what is your windows arm 11 build installed? I started to feel it is windows and not UTM issue.

I have Windows 11 Home Arm64 Build v10.0.22000. It's running very nice and fast so far. I switched back to GPU and have up "suspend" (does not allow this with a GPU enabled). I like "hibernate" much better, because it is actually a Windows feature and not one from the host system/app. One of the apps I was using crashed a couple of times when I used "suspend". It might have to do with it – or not. I have not investigated it further...
